solawi/commands.py

```diff
@@ -34,3 +34,40 @@ def change_password(email):
 @click.option("--interactive/--non-interactive", default=False)
 def import_statements(interactive):
     import_fin_ts(interactive)
+
+
+@app.cli.command()
+@click.argument("person_id")
+@click.argument("share_id")
+@click.argument("date")
+def split_deposits(person_id, share_id, date):
+    """Takes deposits of a given person and creates a new virtual person where
+       all deposits made before `date` are assigned to this person.
+       The existing person is then assigned to deposit into `share_id`.
+       This is to be used in case a person switches their share but continues to deposit money.
+       """
+
+    person = Person.get(person_id)
+
+    virtual_person = Person(name=person.name + " [alt - vom Kontotool erstellt]", share_id=person.share_id)
+    db.session.add(virtual_person)
+
+    person.share_id = share_id
+    db.session.add(person)
+
+    deposits = Deposit.query\
+        .filter(Deposit.person_id == person_id)\
+        .filter(Deposit.timestamp < date)\
```

Comment:
shouldn't we parse the date ?

Reply:
Curiously, we don't have to. I think sqlalchemy (or maybe rather Postgres itself?) takes care of that for us. I have just added a test to ensure that it works as expected with just the string provided.